recipes/systemc/2.3.3/conanfile.py

```diff
@@ -0,0 +1,115 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class SystemcConan(ConanFile):
+    name = "systemc"
+    version = "2.3.3"
+    description = """SystemC is a set of C++ classes and macros which provide
+                     an event-driven simulation interface."""
+    homepage = "https://www.accellera.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    topics = ("simulation", "modeling", "esl", "tlm")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "disable_async_updates": [True, False],
+        "disable_copyright_msg": [True, False],
+        "disable_virtual_bind": [True, False],
+        "enable_assertions": [True, False],
+        "enable_immediate_self_notifications": [True, False],
+        "enable_pthreads": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "disable_async_updates": False,
+        "disable_copyright_msg": False,
+        "disable_virtual_bind":  False,
+        "enable_assertions": True,
+        "enable_immediate_self_notifications": False,
+        "enable_pthreads": False
+    }
+    generators = "cmake"
+    exports_sources = "patches/**"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration("Macos build not supported")
+
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("The compilation of SystemC as a "
+                                            "DLL on Windows is currently not "
+                                            "supported")
+
+        if tools.valid_min_cppstd(self, "17"):
+            raise ConanInvalidConfiguration(
+                "C++ Standard %s not supported by SystemC" %
+                self.settings.compiler.cppstd)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("systemc-{}".format(self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        if self.settings.compiler.cppstd:
+            self._cmake.definitions["CMAKE_CXX_STANDARD"] = \
+                    self.settings.compiler.cppstd
```

Comment:
I just checked my `.conan/settings.yml` and it contains:
```
cppstd: [None, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20]
```
I don't know whether cmake will accept `gnu98`.
Also, maybe you don't need to set `CMAKE_CXX_STANDARD` at all? Maybe conan will pass this flag automatically?
I don't know :smile:

Reply:
Cmake will not like that, from the doc `Supported values are 98, 11, 14, 17, and 20.`

I will see how conan handles the `CMAKE_CXX_STANDARD` but we need to watch out here because of systemc embedding the version into the namespace. If it is not propagated correctly it will fallback to 98 and render the library useless if the compiler assumes another version by default.

Reply:
I just checked, and the `conan_set_std` macro sets the cmake standard.
So unless you pass `SKIP_STD` to `conan_basic_setup`, the standard is set correctly if you use the conan build helper.

So it seems passing `CMAKE_CXX_STANDARD` isn't necessary in neither `conanfile.py`.

Reply:
I came to the same conclusion, thx for double checking.
Updated the two `conanfile.py` files